geps/gep-1867.md

```diff
@@ -0,0 +1,134 @@
+Note the important part of this is the `Gateway` change; the `GatewayClass` aspect is mostly for consistency.
+
+The exact fields are out of scope for this GEP and will be handled by additional GEPs.
+Some example GEPs already depending on this are [GEP-1713](/geps/gep-1713.md), [GEP-1651](/geps/gep-1651.md), and [GEP-1762](/geps/gep-1762.md)
+
+The fields as defined below are, of course, not useful.
+This is intended as a basis for other PRs, not to provide value on its own.
+This GEP will remain in provisional until at least one field is ready to be promoted.
+
+```go
+type GatewaySpec struct {
+  // Infrastructure defines infrastructure level attributes about this Gateway instance.
+  Infrastructure GatewayInfrastructure `json:"infrastructure"`
+  // ...
+}
+
+type GatewayClassSpec struct {
+  // Infrastructure defines infrastructure level attributes for all Gateways in this class.
+  // A Gateway may provide configuration for the same values; as all fields in GatewayInfrastructure are implementation specific,
+  // the merging logic between these is as well. However, the GatewayClass is generally expected to be providing defaults
+  // rather than overrides.
+  Infrastructure GatewayClassInfrastructure `json:"infrastructure"`
+  // ...
+}
+
+type GatewayInfrastructure struct {
+  // ParametersRef provides a arbitrary implementation-specific configuration for
+  // fields not expressed directly in this struct.
+  // This follows the same semantics as GatewayClass's ParametersRef, but lives on the Gateway.
+  ParametersRef ParametersReference
+}
+
+type GatewayClassInfrastructure struct {
+}
+```
+
+### API Principles
+
+For any given field, we will need to make two decisions:
+* whether this should be a first-class field or a generic `parametersRef`.
+* whether this field should be configurable on a Gateway and/or GatewayClass level
+
+The choice to use an extension (`parametersRef`) or first-class field is a well known problem across the API, and the same logic will be used here.
+Fields that are generally portable across implementations and have wide-spread demand and use cases will be promoted to first-class fields,
+while vendor specific or niche fields will remain extensions.
+Because infrastructure is somewhat inherently implementation specific, it is likely most fields will be Extended or ImplementationSpecific.
+However, there are still a variety of concepts that have some meaning between implementations that can provide value to users.
+
+Introduction at Gateway or GatewayClass level will depend on the specific field and use cases for the field.
+In general, it makes sense to provide defaults (GatewayClass) and specific settings (Gateway) for most fields, but
+this will be evaluated on a case-by-case basis.
+
+### Status
+
+The API should likely expose some status. However, it is not yet clear what that will look like.
+This will be addressed prior to promotion beyond "Provisional".
```

Comment:
Can you add an "unresolved" keyword somewhere in here just so we don't miss this prior to promotion?